### ISSUE TITLE
Duplicate variable not detected in function arguments

### ISSUE THREAD

**Issue by wereHamster** [palantir/tslint#320]
I'd expect this to fail the `no-duplicate-variable` check, however tslint silently accepts that:

```javascript
function test(a, a) {
}
```

I came across this when trying to implement a new option to `no-duplicate-variable` which would allow duplicate variables if the name is an underscore. This is releated to #314. When that pull request is accepted then I'll be able to write:

```javascript
function test(_a, b, _c) {
}
```

But sometimes I don't want to name the arguments at all so actually the following would be much nicer:

```javascript
function test(_, b, _) {
}
```

Haskell supports this form natively and in Lua it's common to use a plain underscore for unused variables.

Now you might think why define the function with three arguments when only one is actually used? Well, this is a constructed example, but consider an interface which defines a function with many arguments and the implementation only uses some.

**Comment by adidahiya** [comment]
this is fixed in the latest tslint.

**Comment by wereHamster** [comment]
I dont see any test covering that in `test/files/rules/duplicate-variable.test.ts`.

**Comment by wereHamster** [comment]
Hm, indeed there is (line 98, `function testArguments1`).

**Comment by wereHamster** [comment]
For reference, fixed by 7642223b6 (no-duplicate-variable: treat function arguments as vars).
